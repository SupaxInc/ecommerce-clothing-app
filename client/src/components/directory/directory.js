import React from "react";
import MenuItem from "../menu-item/menu-item";

import { useSelector } from "react-redux";
import { selectDirectorySections } from "../../redux/directory/directory.selector";

import './directory.scss';


const Directory = () => {
    const sections = useSelector(selectDirectorySections)
    return (
        <div className="directory-menu">
            {
                sections.map(({ id,...otherProps }) => {
                    // Object spreading below of ...otherProps is similar to title={title} imageUrl={imageUrl} etc...
                    return <MenuItem key={id} {...otherProps} />
                })
            }
        </div>
    );
}


export default Directory;