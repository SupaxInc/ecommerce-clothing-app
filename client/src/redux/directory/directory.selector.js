import { createSelector } from 'reselect';

// Select the directory state from our Store
export const selectDirectory = state => state.directory;

// Select the sections state from our directory reducer
export const selectDirectorySections = createSelector(
    [selectDirectory],
    (directory) => directory.sections
);