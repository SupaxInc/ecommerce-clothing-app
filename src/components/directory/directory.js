import React, { Component } from "react";
import './directory.scss';
import MenuItem from "../menu-item/menu-item";
import { sections } from "./directory.data";

class Directory extends Component {
    constructor() {
        super();
        this.state = {
            sections: sections
        }
    }

    render() {
        const { sections } = this.state;
        return (
            <div className="directory-menu">
                {
                    sections.map(({ id,...otherProps }) => {
                        // Object spreading below of ...otherProps is similar to title={title} imageUrl={imageUrl} etc...
                        return <MenuItem key={id} {...otherProps} />
                    })
                }
            </div>
        )
    }

}

export default Directory;