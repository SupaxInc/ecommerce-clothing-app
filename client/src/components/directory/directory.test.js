import React from "react";
import { shallow } from "enzyme";

import {sections} from '../../redux/directory/directory.data';

import * as redux from 'react-redux';
import Directory from "./directory";
import MenuItem from "../menu-item/menu-item";

describe('Directory component', () => {
    let wrapper;

    // Mock selector hook
    const useSelectorMock = jest.spyOn(redux, 'useSelector');
    const mockState = {
        directory: {
            sections
        }
    }

    beforeEach(() => {
        useSelectorMock.mockImplementation(callback => callback(mockState));
        wrapper = shallow(<Directory />);
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should render directory', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the amount of sections from the sections array', () => {
        expect(wrapper.find(MenuItem).length).toBe(5);
    });
}); 