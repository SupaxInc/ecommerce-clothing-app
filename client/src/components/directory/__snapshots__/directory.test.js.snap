// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory component should render directory 1`] = `ShallowWrapper {}`;
